src/infrastructure/validation/testing/TestHarness.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * @author        Abhishek Saran, Manas Sanjay
+ * @module        Infrastructure module 
+ * @team          Test Harness
+ * @description   Test Harness main class
+ * @summary       To provide functionality to tester to run test case sets in various ways- 
+                  Methods such as runAll(), runByName(), runByCategroy() and run by  Priority() are provided. 
+ */
+package infrastructure.validation.testing;
+import java.io.File;
+import java.util.*;
+
+public class TestHarness{
+  
+  private String savePath;
+  
+  void setSavePath(String path){
+    this.savePath=path;
+  }
+  
+  String getSavePath(){
+    return this.savePath;
+  }
+  
+  /**
+  * static helper method to return all the test cases of project
+  * @return    ArrayList of File type containing all the test cases with valid naming
+  */
+  private static ArrayList<File> getAllTests(){
+    File[] modules = new File("../../../").listFiles(File::isDirectory);
+    ArrayList<File> allTests = new ArrayList<File>();
+		for(File module : modules){
+		  if(module.isDirectory()){
+		  	String strModule = module.getName();
+		  	System.out.println(strModule);
+			  try {
+			    File f = new File("src/"+strModule+"/tests/");
+				  File[] tests  = f.listFiles(File::isFile);
+				  for (File test : tests){
+					  if(test.getName().endsWith("Test.java")){
+						  allTests.add(test); 
+					  }
+				  } 
+	    	}
+	    	catch (Exception e){
+	  		  System.out.println("Exception in module: "+strModule+" >> "+e);
+		    }
+		  }
+    } 
+    return allTests;
+  }
+
+
+ /**
+  * Void method to run test case of a given category
+  * @param Category    The name of the category of tests to be run
+  */
+  public void runByCategory(String Category){
+    String path = "../../../" + Category + "/tests";
+    File directoryPath = new File(path);
+    //get all the test cases in tests directory of respective category 
+    String tests[] = directoryPath.list();
+
+    //create object of each test case class to run and get the result of each test
+    for(int i =0; i<tests.length; i++){
+      String[] arrOfStr = tests[i].split(".", 2); 
+      String testClassName = arrOfStr[0];
+      Class<?> testClass = Class.forName("testClassName");
+      Object test = testClass.getDeclaredConstructor().newInstance();
+      if(Category.equals(test.getCategory())){
+        boolean result = test.run();
+      }
+    }
```

Comment:
Can you change his functions to match your pattern and logic. Like using the getAllTests() and the variable naming

Reply:
I can put a try and catch block in case they give wrong category name or tests folder is empty. Otherwise, it's same as my logic. Getting all tests rather than just one category will be extra work.

Reply:
Ok